### DISCUSSION TITLE
Rank mismatch error with brew netcdf install

### DISCUSSION THREAD

**Discussion by ylserra** [Homebrew/discussions#2310] [Casks]
Hi - hope I selected the right category. I have a question about the netcdf install from Homebrew. Specifically the fortran libraries.

I've just updated to Big Sur so had to update my netcdf, gfortran, ncar-ncl, and udunits libraries. I used to use MacPorts for all but the gfortran, which was only available from brew for Catalina at the time I installed all this stuff. Anyway, my current system is only using brew (/usr/local) and I've removed the Macports installs (/opt/local). After installing I tried to compile an old f77 code that calls udunits and netcdf libraries. This code compiled using gfortran just a week ago when my system was Catalina based. Now I get the following error:

yserra % gfortran -L./ -I./ -I/usr/local/include stfilt_olrnc.f handle_err.f get_time.f libfilt.a -o stfilt_olrnc -L/usr/local/lib -lnetcdff -ludunits
stfilt_olrnc.f:433:43:

  401 |       status=nf_get_vara_real(fid,ivarid,1,NLAT,rlat)
      |                                         2  
......
  433 |          status=nf_get_vara_real(fid,varid,starti,counti,xxr)
      |                                           1
Error: Rank mismatch between actual argument at (1) and actual argument at (2) (scalar and rank-1)
stfilt_olrnc.f:713:55:

  401 |       status=nf_get_vara_real(fid,ivarid,1,NLAT,rlat)
      |                                         2              
......
  713 |                   status=nf_get_vara_real(foutid,varid,start,count,
      |                                                       1
Error: Rank mismatch between actual argument at (1) and actual argument at (2) (scalar and rank-1)
stfilt_olrnc.f:739:53:

  401 |       status=nf_get_vara_real(fid,ivarid,1,NLAT,rlat)
      |                                         2            
......
  739 |      $                 nf_get_vara_real(foutid,varid,start,count,
      |                                                     1
Error: Rank mismatch between actual argument at (1) and actual argument at (2) (scalar and rank-1)
 
varid is declared once as an integer. Not sure why the same subroutine (nf_get_vara_real) seems to think it has two different types.

Thanks for any help.

**Comment by ylserra** [comment]
I just answered my own question. I guess the newer netcdf fortran library does not let me send a number to this function and then later use a variable (declared as an integer). By using
start1=1
count1=NLAT 
and putting start1 and count1 into line 401 I avoided this error. Not sure I understand why but it worked. 

My next issue is that the brew installation of udunits is really only udunits2 but my code requires udunits. 

Any hope in getting udunits cask for brew? :)

**Reply by SMillerDev** [reply]
> Any hope in getting udunits cask for brew? :)

You can always submit a version to `homebrew-cask-versions`

**Reply by ylserra** [reply]
Ah, well, if I could get it to compile on Big Sur I'd be happy to share but for now, no luck.
